src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,50 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Track;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            List<Bulldozer> bulldozers = new BulldozerProducer().get();
+            for (int i = 0; i < bulldozers.size(); i++) {
+                machines.add(bulldozers.get(i));
+            }
```

Comment:
you can replace this with
```suggestion
            machines.addAll(bulldozers);
```